client-spark/shuffle-manager-2/src/main/scala/org/apache/spark/shuffle/rss/RssShuffleManager.scala

```diff
@@ -61,7 +62,12 @@ class RssShuffleManager(conf: SparkConf) extends ShuffleManager with Logging {
         if (lifecycleManager.isEmpty) {
           val metaSystem = new LifecycleManager(appId, rssConf)
           lifecycleManager = Some(metaSystem)
-          rssShuffleClient = Some(ShuffleClient.get(metaSystem.self, rssConf))
+          rssShuffleClient = Some(ShuffleClient.get(
+            metaSystem.self,
+            rssConf,
+            lifecycleManager.map(_.getUserIdentifier()).getOrElse(UserIdentifier(
```

Comment:
lifecycleManager is guaranteed to have value here

Reply:
Yea